src/Delivery/DynamicEntry.php

```diff
@@ -130,7 +136,15 @@ public function __call($name, $arguments)
         }
 
         if ($result instanceof Link) {
-            return $client->resolveLink($result);
+            $cacheId = $result->getLinkType() . '-' . $result->getId();
+            if (isset($this->resolvedLinks[$cacheId])) {
+                return $this->resolvedLinks[$cacheId];
+            }
+
+            $resolvedObj = $client->resolveLink($result);
```

Comment:
I wouldn't issue an extra request for this, and rather leave the link as is.

This is specially the case for deleted referenced entries/assets.